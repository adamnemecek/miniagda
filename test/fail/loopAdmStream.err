MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "loopAdmStream.ma" ---
--- scope checking ---
--- type checking ---
type  Unit : Set
term  unit : Unit
type  Unit : Set
type  Stream : ++(A : Set) -> - Size -> Set
term  cons : .[A : Set] -> .[i : Size] -> ^(head : A) -> ^(tail : Stream A i) -> Stream A $i
type  Stream : ++(A : Set) -> - Size -> Set
G> fun head : .[A : Set] -> .[i : Size] -> (__cons : Stream A $i) -> A
G> { head A i (cons .A .i __head __tail) = __head
}
term  head : .[A : Set] -> .[i : Size] -> (__cons : Stream A $i) -> A
{ head [A] [i] (cons [.A] [.i] __head __tail) = __head
}
G> fun tail : .[A : Set] -> .[i : Size] -> (__cons : Stream A $i) -> Stream A i
G> { tail A i (cons .A .i __head __tail) = __tail
}
term  tail : .[A : Set] -> .[i : Size] -> (__cons : Stream A $i) -> Stream A i
{ tail [A] [i] (cons [.A] [.i] __head __tail) = __tail
}
term  f : (Stream Unit # -> Stream Unit #) -> .[i : Size] -> (Stream Unit i -> Stream Unit #) -> Stream Unit i
error during typechecking:
f
/// clause 1
/// pattern $j
/// checkPattern $j : matching on size, checking that target .[i : Size] -> (~2 : (~3 : Stream Unit i) -> Stream Unit #) -> Stream Unit i ends in correct coinductive sized type
/// new i : Size
/// endsInSizedCo: (~2 : (~3 : Stream Unit i) -> Stream Unit #) -> Stream Unit i
/// new  : ((Stream {Unit {i = v1}} v1)::Tm -> Stream Unit #{i = v1})
/// type (~3 : Stream Unit i) -> Stream Unit # not lower semi continuous in i
