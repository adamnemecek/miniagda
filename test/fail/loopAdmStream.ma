-- 2010-05-11

data Unit : Set
{ unit : Unit 
}

sized codata Stream (+ A : Set) : Size -> Set 
{ cons : [i : Size] -> (head : A) -> (tail : Stream A i) -> Stream A ($ i)
}
fields head, tail
 
-- the type of f is not admissible
cofun f : (Stream Unit # -> Stream Unit #) ->
  (i : Size) -> (Stream Unit i -> Stream Unit #) -> Stream Unit i
{ f h ($ j) g = 
    h (g (cons j unit (f (\ x -> h (h x)) j (\ x -> g (cons j unit x))))) 
}

let bla : Stream Unit # = f (tail Unit #) # (\ x -> x)

-- LOOP!
eval let u : Unit = head Unit # bla
eval let us : Stream Unit # = tail Unit # bla
