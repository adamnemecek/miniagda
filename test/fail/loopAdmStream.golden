MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "loopAdmStream.ma" ---
--- scope checking ---
--- type checking ---
type  Unit : Set
term  Unit.unit : < Unit.unit : Unit >
type  Stream : ++(A : Set) -> - Size -> Set
term  Stream.cons : .[A : Set] -> .[i : Size] -> ^(head : A) -> ^(tail : Stream A i) -> < Stream.cons i head tail : Stream A $i >
term  head : .[A : Set] -> .[i : Size] -> (cons : Stream A $i) -> A
{ head [A] [i] (Stream.cons [.i] #head #tail) = #head
}
term  tail : .[A : Set] -> .[i : Size] -> (cons : Stream A $i) -> Stream A i
{ tail [A] [i] (Stream.cons [.i] #head #tail) = #tail
}
term  f : (Stream Unit # -> Stream Unit #) -> .[i : Size] -> (Stream Unit i -> Stream Unit #) -> Stream Unit i
error during typechecking:
f
/// clause 1
/// pattern $j
/// checkPattern $j : matching on size, checking that target .[i : Size] -> (Stream Unit i -> Stream Unit #) -> Stream Unit i ends in correct coinductive sized type
/// new i <= #
/// endsInSizedCo: (Stream Unit i -> Stream Unit #) -> Stream Unit i
/// type  Stream Unit i -> Stream Unit #  not lower semi continuous in  i
